ports/psoc6/Makefile

```diff
@@ -81,77 +95,63 @@ CXXFLAGS += $(CXXFLAGS_MOD)
 
 # Flags for user C modules
 CFLAGS  += $(CFLAGS_MOD)
-CFLAGS  += -DFFCONF_H=\"lib/oofatfs/ffconf.h\" -DLFS1_NO_MALLOC -DLFS1_NO_DEBUG -DLFS1_NO_WARN -DLFS1_NO_ERROR -DLFS1_NO_ASSERT -DLFS2_NO_MALLOC -DLFS2_NO_DEBUG -DLFS2_NO_WARN -DLFS2_NO_ERROR -DLFS2_NO_ASSERT
+
+# Flags from extmod
+CFLAGS  += $(CFLAGS_THIRDPARTY) $(CFLAGS_EXTMOD)
+
 
 LDFLAGS += $(LDFLAGS_MOD)
 
 LIBS =
 
 
-MPY_SRC_C += lib/littlefs/lfs1.c lib/littlefs/lfs1_util.c lib/littlefs/lfs2.c lib/littlefs/lfs2_util.c \
-		 	 lib/oofatfs/ff.c lib/oofatfs/ffunicode.c \
-		 	 \
-		 	 shared/readline/readline.c \
-			 \
-			 shared/runtime/gchelper_native.c \
-			 shared/runtime/interrupt_char.c \
-		 	 shared/runtime/pyexec.c \
-		 	 shared/runtime/stdout_helpers.c \
-			 \
-		 	 shared/timeutils/timeutils.c \
-		     \
-			 shared/runtime/sys_stdio_mphal.c 
-
+MPY_SRC_C = shared/readline/readline.c \
+			\
+			shared/runtime/gchelper_native.c \
+			shared/runtime/interrupt_char.c \
+		 	shared/runtime/pyexec.c \
+		 	shared/runtime/stdout_helpers.c \
+			\
+		 	shared/timeutils/timeutils.c \
+		    \
+			shared/runtime/sys_stdio_mphal.c \
+			\
+			help.c \
+			\
+    		drivers/machine/psoc6_gpio.c \
+    		drivers/machine/psoc6_i2c.c \
+    		drivers/machine/psoc6_system.c  \
+			\
+			modules/gc/modgc.c \
+			\
+			modules/machine/modmachine.c \
+			modules/machine/machine_i2c.c \
+			modules/machine/machine_pin.c \
+			modules/machine/machine_rtc.c \
+			modules/machine/pins.c \
+			\
+			modules/psoc6/modpsoc6.c \
+			modules/psoc6/psoc6_fatfs.c \
+			modules/psoc6/psoc6_flash.c \
+			modules/psoc6/psoc6_qspi_flash.c \
+			\
+			modules/time/modutime.c \
+			\
+			$(SRC_THIRDPARTY_C) \
+			$(SRC_USERMOD) \
+			$(SRC_EXTMOD_C)
+		 	 
 
 SRC_ASM = shared/runtime/gchelper_thumb1.s
 
 
-SRC_C = \
- 	main.c \
-	\
-	help.c \
-	mphalport.c \
-	\
-    drivers/machine/psoc6_gpio.c \
-    drivers/machine/psoc6_i2c.c \
-    drivers/machine/psoc6_system.c  \
-	\
-	modules/gc/modgc.c \
-	\
-	modules/machine/modmachine.c \
-	modules/machine/machine_i2c.c \
-	modules/machine/machine_pin.c \
-	modules/machine/pins.c \
-	modules/machine/machine_rtc.c \
-	\
-	modules/psoc6/modpsoc6.c \
-	modules/psoc6/psoc6_fatfs.c \
-	modules/psoc6/psoc6_flash.c \
-	modules/psoc6/psoc6_qspi_flash.c \
-	\
-	modules/time/modutime.c \
-	\
-	$(MPY_SRC_C) \
-    \
-	$(BUILD)/frozen_content.c
-
-
-SRC_C    += $(SRC_USERMOD)
+SRC_C    += main.c mphalport.c $(MPY_SRC_C) $(BUILD)/frozen_content.c
 SRC_CXX   = 
 
 
-SRC_QSTR += $(MPY_SRC_C) $(SRC_USERMOD) $(SRC_EXTMOD_C)
-SRC_QSTR += modules/machine/modmachine.c \
-            modules/machine/machine_i2c.c \
-            modules/machine/machine_pin.c \
-			modules/machine/machine_rtc.c \
+SRC_QSTR += $(MPY_SRC_C) \
             \
-            modules/os/moduos.c \
-			\
-            modules/psoc6/modpsoc6.c modules/psoc6/psoc6_fatfs.c modules/psoc6/psoc6_flash.c modules/psoc6/psoc6_qspi_flash.c \
-			\
-			modules/time/modutime.c
-
+            modules/os/moduos.c
```

Comment:
Why this one is not with the rest of MPY_SRC_C?

Reply:
Included though mpconfigport.h (see above).